### ISSUE TITLE
Error: While importing "microblog", an ImportError was raised:

### ISSUE THREAD

**Issue by RamanSB** [miguelgrinberg/microblog#158]
Hey Miguel, I downloaded the code from chapter 15 (the download link) where you introduced blueprints and application factory. I moved the downloaded folder in to another directory on my local machine. I then created a venv (python3 -m venv venv) and activated it (source venv/bin/activate) I then pip installed all flask extensions.

I also set the FLASK_APP variable to be 'microblog.py' - however everytime run [flask run]:

I get the following exception:

> 
> flask.cli.NoAppException
> flask.cli.NoAppException: While importing "microblog", an ImportError was raised:
> 
> Traceback (most recent call last):
>   File "/Users/RamanSB/anaconda3/lib/python3.7/site-packages/flask/cli.py", line 235, in locate_app
>     __import__(module_name)
>   File "/Users/RamanSB/Documents/Python/Python Lesuire/Flask/MegaTutorial/microblog/microblog.py", line 1, in <module>
>     from app import create_app, db, cli
>   File "/Users/RamanSB/Documents/Python/Python Lesuire/Flask/MegaTutorial/microblog/app/__init__.py", line 5, in <module>
>     from flask_sqlalchemy import SQLAlchemy
> ModuleNotFoundError: No module named 'flask_sqlalchemy'
> 

It doesn't matter if I re-order the imports the extension imported first will always be mentioned in the ModuleNotFoundError, I'm not sure but I think this may be a circular dependency issue or something to do with my venv.

Your help will be greatly appreciated, if you need any more information please let me know.

**Comment by miguelgrinberg** [comment]
Did you install the dependencies in a virtualenv?

**Comment by lucasmbarboza** [comment]
Hey, Miguel. 
I'm facing this erro on project based on microblog
Error: While importing "tcc-python3", an ImportError was raised:

Traceback (most recent call last):
  File "/home/lucas/tcc-python3/venv/lib/python3.6/site-packages/flask/cli.py", line 240, in locate_app
    __import__(module_name)
  File "/home/lucas/tcc-python3/tcc-python3.py", line 1, in <module>
    from app import app
  File "/home/lucas/tcc-python3/app/__init__.py", line 3, in <module>
    from config import Config
ModuleNotFoundError: No module named 'config'

**Comment by miguelgrinberg** [comment]
@lucasmbarboza well, do you have a config.py in the current directory?

**Comment by lucasmbarboza** [comment]
Hey! It was a problem on migration of my project from python 2.7 to 3.6 the
sintax of os.path() changed, now instead of use __file__ have use
 "__file__".


Em sex, 25 de out de 2019 às 19:22, Miguel Grinberg <
notifications@github.com> escreveu:

**Comment by Sean-Horner-SmoothStack** [comment]
Howdy Miguel,

I know this project was a while ago, but I'm also getting an ImportError that is driving me up a wall. I'm on section 3 Webforms, and everything was running great in the last two sections, but now whenever I try to use 'flask run' I get an ImportError being rasied with absolutely no traceback. I have tried pretty much everything, I searched all my files for erroneous imports, I replaced every file in my project with yours from the Github at that step, and I made sure I'm in my virtual environment with the two dependencies need for this stage (flask and wtforms) and nothing. This seems super odd that I would trigger an error with absloutely no traceback, any suggestions?
